### PR TITLE
Porting 8.0 product_customer_code

### PR DESCRIPTION
**Pull request by hurrinico**
I've made a fork of https://github.com/OCA/product-attribute/pull/38 and rebase it

### CODE REVIEW

**Comment by eLBati** [comment]
Please see travis

```
product_customer_code/product.py:27:1: F401 '_' imported but unused
product_customer_code/product_customer_code.py:27:1: F401 '_' imported but unused
```

**Comment by hurrinico** [comment]
@eLBati fixed

**Comment by eLBati** [comment]
@hurrinico please consider suggestions made since https://github.com/OCA/product-attribute/pull/38#discussion-diff-22769482

File: product_customer_code/product.py [hunk 1]
```diff
@@ -24,7 +24,6 @@
 #
```
Comment (eLBati) [inline product_customer_code/product.py]: Not imported python files should be removed

**Comment by pedrobaeza** [comment]
I think a better path is to use _product_supplierinfo_for_customer_ module as base for the customer code instead of creating a new model.

**Comment by eLBati** [comment]
We are using `product_customer_code` in several 7.0 and 8.0 instances and I would prefer to keep the current structure, to avoid issues with the existing databases

**Comment by pedrobaeza** [comment]
OK, but we should merge both concepts in v9.

**Comment by andhit-r** [comment]
@hurrinico would you:
1. Adjust README.rst according to latest OCA guideline
2. Use simple header on every .py file

**Comment by mikevhe18** [comment]
I've made some changes to increase coverage/coveralls, then also adjustments to version 8 and OCA guidelines.

**Comment by eLBati** [comment]
@hurrinico @mikevhe18  please see failing checks

```
************* Module product_customer_code
product_customer_code/__init__.py:1: [W7908(missing-newline-extrafiles), ] views/product_product_view.xml Missing newline
************* Module product_customer_code.__openerp__
product_customer_code/__openerp__.py:5: [C8106(manifest-version-format), ] Wrong Version Format "1.0" in manifest file. Regex to match: "(\d+.\d+.\d+.\d+.\d+)"
```

**Comment by mikevhe18** [comment]
hello @hurrinico, i've fixed the code. Thanks in advance

**Comment by andhit-r** [comment]
Tested functionality on runbot :+1:

**Comment by mikevhe18** [comment]
:+1:

**Comment by andhit-r** [comment]
Hi @OCA/product-maintainers would you kindly review this PR? Many thanks

**Comment by mikevhe18** [comment]
Hello @pedrobaeza @eLBati, would you kindly review this PR again ?

**Comment by dcoopennet** [comment]
:+1:

File: product_customer_code/README.rst [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+   :alt: License: AGPL-3
+    
+======================
+Products Customer Code
+======================
+
+This module does just like the product.supplierinfo but for customers instead. For instance it
+allows to have different references for the same product according to the customer.
+
+.. image:: product_customer_code/static/src/img/screenshot1.png
```
Comment (eLBati) [inline product_customer_code/README.rst]: Please use figure like at https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/README.rst

File: product_customer_code/README.rst [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+===========
+
+Bugs are tracked on `GitHub Issues
+<https://github.com/OCA/product-attribute/issues>`_. In case of trouble, please
+check there if your issue has already been reported. If you spotted it first,
+help us smashing it by providing a detailed and welcomed `feedback
+<https://github.com/OCA/
+product-attribute/issues/new?body=module:%20
+product_customer_code%0Aversion:%20
+8.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (eLBati) [inline product_customer_code/README.rst]: Please use the new bug tracker section like at https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/README.rst

**Comment by eLBati** [comment]
A few comments then LGTM :+1:

**Comment by hurrinico** [comment]
@eLBati i've made the requested fix

**Comment by bealdav** [comment]
tested LGTM for me :+1: 
off-topic Migration: in product.product view: 'Customer Code' tab name could replaced by 'Customer Infos' because other modules could extend custom informations (as name or any field)

**Comment by pedrobaeza** [comment]
Please check Travis status. Do we need to have a specific task for this?

**Comment by eLBati** [comment]
@pedrobaeza , @espo-tony is on the problem of `duplicate key value violates unique constraint "product_template_uniq_prefix_code"`

**Comment by espo-tony** [comment]
@pedrobaeza , @eLBati 
Travis seems to raise an error while running a tes. More specifically, it returns a `duplicate key value violates unique constraint "product_template_uniq_prefix_code"` after invoking the copy method on the model product.product.

According to the tests and the analysis I performed, the issue doesn't look to depend on the module "product_customer_code". Instead, it looks like the duplication of a product.product record is not possible anymore because of a bug introduced by the modules 'product_code_builder' and 'product_code_builder_sequence'.

Please give a look to the PRs #181 , #182 for further informations on this bug and to review a possible solution to them (notice that solving those bugs, also the Travis errors remarked on this PR should be solved).

**Comment by eLBati** [comment]
For me this can be closed , as ```product_supplierinfo_for_customer``` is already available and overlapping with this.
See https://github.com/OCA/product-attribute/pull/236#issuecomment-304812958

**Comment by hurrinico** [comment]
IMHO this can be closed

**Pull request closed without merging**